trellowarrior/main.py

```diff
@@ -16,454 +17,708 @@
-                # Do sync Trello - Task Warrior
-                sync_trello_tw(trello_lists,
-                        project['tw_project_name'],
-                        project['trello_board_name'],
-                        project['trello_todo_list'],
-                        project['trello_doing_list'],
-                        project['trello_done_list'])
-                # Upload new Task Warrior tasks
-                upload_new_tw_tasks(trello_lists,
-                        project['tw_project_name'],
-                        project['trello_board_name'],
-                        project['trello_todo_list'],
-                        project['trello_doing_list'],
-                        project['trello_done_list'])
+        projects = None
+    tw.sync(projects)
+    seconds = args.delai_between_sync * 60
+    if seconds > 0:
+        try:
+            while True:
+                logger.info("Sync done. will wait {} minutes before next one "
+                            "(press ctrl-c to stop)".format(args.delai_between_sync))
+                sleep(seconds)
+                tw.invalid_trello_cache()
+                tw.sync(projects)
+        except KeyboardInterrupt:
+            pass
 
 def authenticate(args):
     """
     Function called by the authenticate subcommand to perform the initial authentication
 
     :args: Command line arguments. Must contain the following attributes : 'config', 'api_key', 'api_key_secret', 'name'
     """
-    if os.path.exists(args.config):
-        print("File already exists, aborting")
-
-    oauth_token = create_oauth_token(expiration=args.expiration, key=args.api_key, secret=args.api_key_secret, name=args.name)
-    conf = RawConfigParser()
-    conf.set("DEFAULT", "trello_api_key", args.api_key)
-    conf.set("DEFAULT", "trello_api_secret", args.api_key_secret)
-    conf.set("DEFAULT", "trello_token", oauth_token['oauth_token'])
-    conf.set("DEFAULT", "trello_token_secret", oauth_token['oauth_token_secret'])
-    with open(args.config, 'w') as f:
-        conf.write(f)
+    tw = TrelloWarrior(config_file=args.config)
+    tw.authenticate(args.expiration, args.api_key, args.api_key_secret, args.name)
+
+
+def new_project(args):
+    """
+    Function called by the new subcommand to add a project to the config
+
+    :args: Command line arguments. Must contain the following attributes : 'config', 'name', 'tw_name', 'board_name',
+            'todo', 'doing', 'done', 'no_sync'
+    """
+    tw = TrelloWarrior(config_file=args.config)
+    tw.new_project(args.name, args.tw_name, args.board_name, args.todo, args.doing, args.done, args.no_sync)
+
 
 def main():
     parser = argparse.ArgumentParser(description="Bidirectional sync between trello and taskwarrior")
     parser.add_argument('-c', '--config', metavar='value', help='custom configuration file path')
+    parser.add_argument('-v', '--verbose', help='Increase verbosity. Can be repeated up to 2 times', action='count')
```

Comment:
I think that we can add here the following line.
```python
parser.set_defaults(func=sync)
```
With this the user can exec `trellowarrior` directly to do `sync`

Reply:
This will not work since with sub_parser the subcommand is mandatory. It is possible to work around this limitation, but it is not trivial.

Reply:
It works but only in Python3 since you can define `required` (that is false by default) as `subparsers.required = True` or `subparsers.required = False`.
See:
https://github.com/python/cpython/pull/3027
For Python2 is as you say, it required a difficult workaround and that not worth it.